experimental/handoff-library/csharp_dotnetcore/Microsoft.Bot.Builder.Handoff/TurnContextExtender.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Handoff
+{
+    /// <summary>
+    /// Extends ITurnContext to add InitiateHandoffAsync.
+    /// </summary>
+    public static class TurnContextExtender
+    {
+        /// <summary>
+        /// Initiate handoff to human agent.
+        /// </summary>
+        /// <param name="turnContext">Turn context.</param>
+        /// <param name="activities">Transcript of the activities that took place so far.</param>
+        /// <param name="handoffContext">Additional channel-specific content.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>IHandoffRequest.</returns>
+        public static Task<IHandoffRequest> InitiateHandoffAsync(this ITurnContext turnContext, IActivity[] activities, object handoffContext, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var adapter = turnContext.Adapter;
+            if (adapter is IHandoffAdapter handoffAdapter)
+            {
```

Comment:
Love this. Please just have this be an argument check + throw on the way in. 

That way the bulk of the method code isn't in an if statement, which makes it easier to follow.